pubsub/redis_publisher.py

```diff
@@ -11,7 +11,7 @@
 
 class RedisPublisher(Publisher):
 
-    def __init__(self, connection: RedisConnection = RedisConnection(),
+    def __init__(self, connection: RedisConnection,
                  publisher_thread_pool: ThreadPoolExecutor = ThreadPoolExecutor(max_workers=4)):
```

Comment:
Do you think the other defaults will also not be honoured? Like the thread_pool?

Reply:
I think the issue here would only occur because of this default since it uses ```self.connect()```  init which would throw an exception when connection is not successful. Other default arguments do not share the same issue.

Reply:
Ok great.